### PR TITLE
Pull request with all files together

### PR DESCRIPTION
**Pull request by bquiring**
by request

### CODE REVIEW

File: codes/classical/bits/convolutional.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: convolutional
+physical: bits
+logical: bits
+
+name: 'Convolutional'
```
Comment (valbert4) [inline codes/classical/bits/convolutional.yml:10]: ...code

File: codes/classical/bits/convolutional.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+introduced: '\cite{manual:{Peter Elias. Coding for noisy channels. IRE Convention Records, 3(4):37–46, 1955.}}'
+
+description: 'Convolutional codes are classical codes that are formed using generator polynomials over the finite field with two elements. The encoder slides across contiguous subsets of the input bit-string (like a convolutional neural network) evaluating the polynomials on that window to obtain a number of parity bits. These parity bits are the encoded information. There are many ways to formulate these codes'
+
+protection: 'Bit flip errors'
```
Comment (valbert4) [inline codes/classical/bits/convolutional.yml:16]: doesn't need to be here

File: codes/classical/bits/convolutional.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+features:
+
+  rate: 'Depends on the polynomials used. Using puncturing \cite{doi:10.12928/telkomnika.v10i4.171} removal the rate for the code can be increased from 1/n to m/n for some m, n.'
```
Comment (valbert4) [inline codes/classical/bits/convolutional.yml:20]: Using puncturing removal \cite{...}, the rate...    ?

File: codes/classical/bits/convolutional.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  encoders: 
+    - 'Evaluation on the generator polynomials. Can be implemented with a small number of XOR gates'
+
+  decoders:
+    - 'Decoders based on the Viterbi algorithm (trellis decoding) were developed first, which result in the most-likely codeword for the encoded bits \cite{doi:10.1109/TIT.1967.1054010}. Following, other trellis decoders such as the BCJR decoding algorithm \cite{doi:10.1109/TIT.1974.1055186} were developed later.'
+
+realizations:
+  - 'A type of convolutional code used in Real-time Applcation networks \cite{manual:{Salehe Mrutu, Anael Sam, Nerey Myungi. Forward Error Correction Convolutional Codes for RTAs’ Networks: An Overview. 2014.}}'
```
Comment (valbert4) [inline codes/classical/bits/convolutional.yml:29]: doi:10.5815/ijcnis.2014.07.03 doesn't work?

File: codes/classical/bits/convolutional.yml (hunk 3 above)
Comment (valbert4) [reply codes/classical/bits/convolutional.yml:20]: Put m,n in \\( ...  \\). Also, n is not the number of bits here, correct? Maybe change the notation here and use different letters for these so as not confused quantum people.

File: codes/classical/bits/convolutional.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  - 'Mobile and radio communications (3G networks) use convolutional codes concatenated with Reed-Solomon codes to obtain suitable performance \cite{manual:{Halonen, Timo, Javier Romero, and Juan Melero, eds. GSM, GPRS and EDGE performance: evolution towards 3G/UMTS. 2004.}}'
```
Comment (valbert4) [inline codes/classical/bits/convolutional.yml:30]: doi:10.1002/0470866969 ?

File: codes/quantum/qubits/permutation_invariant.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+logical: qubits # also qudits
+
+name: 'Permutation Invariant'
+
+introduced: '\cite{arXiv:quant-ph/0304153}'
+
+description: 'Codes which are stabilized by the symmetric group \(S_n\) on \(n\) elements, in a generalization of stabilizer codes to binary codes utilizing (non-abelian) group actions (in particular, \(S_n\) is non-abelian).'
+
+protection: 'Depends on the family. One family (parameterized by \(t\)) protects against arbitrary weight \(t\) qubit errors, another can approximately correct spontaneous decay errors. \cite{arXiv:1302.3247} \cite{doi:10.1103/PhysRevA.93.042340}. Another can protect against amplitude damping \cite{doi:10.1109/TIT.2019.2956142}. Another yet can protect against (multiple) deletion errors \cite{doi:2102.02494} TODO:CITE' # TODO: I would like to cite \cite{doi:2102.03015} (https://arxiv.org/abs/2102.03015) but I keep getting the error 'requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.crossref.org/works/2102.03015/transform/application/vnd.citationstyles.csl+json'
+
```
Comment (valbert4) [inline codes/quantum/qubits/permutation_invariant.yml:17]: Use \cite{arxiv:2102.03015}

File: codes/quantum/qubits/permutation_invariant.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: permutation_invariant
+physical: qubits
+logical: qubits # also qudits
+
+name: 'Permutation Invariant'
```
Comment (valbert4) [inline codes/quantum/qubits/permutation_invariant.yml:10]: code

File: codes/quantum/qubits/permutation_invariant.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+  decoders:
+   - 'For a family of codes, using projection, probability amplitude rebalancing, and gate teleportation can be done in \(O(N^2)\) \cite{arXiv:2102.02494}.'
+
+notes:
+  - 'can be constructed using real polynomials for high-dimensional qudit spaces \cite{doi:10.1016/j.laa.2017.06.031}'
+
+relations:
+  parents:
+    - code_id: stabilizer
```
Comment (valbert4) [inline codes/quantum/qubits/permutation_invariant.yml:33]: add quantum_cyclic as cousin, mentioning that cyclic group is subgroup of stabilizer group

File: codes/quantum/qubits/permutation_invariant.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+#######################################################
```
Comment (valbert4) [inline codes/quantum/qubits/permutation_invariant.yml:1]: This is a good umbrella entry for perm.-inv codes. What I had in mind was actually a specific entry for Ouyang's codes arXiv:1302.3247 (name "GNU permutation-invariant code"). In there, you can express them in terms of Dicke states (we have a way to do display equations now), mention the realization, and also mention the connections found in an appx of https://arxiv.org/abs/1708.05010. I can help fill this in, but was hoping you could give it a shot with a reasonably thorough initial draft.

File: codes/quantum/qubits/quantum_reed_muller.yml [hunk 10]
```diff
@@ -5,26 +5,45 @@
...
+description: 'Quantum Reed-Muller codes are formed from their classical Reed-Muller counterparts in which polynomials over finite fields encode data. This is done by transforming these polynomials into the stabilizer generator matrices'
+
+protection: 'Protects against Pauli noise.'
+
+features:
+
+  rate: '\(\frac{k}{n}\), where \(k = 2^r - {r \choose t} + 2 \sum_{i=0}^{t-1} {r \choose i}\).'
+
+  encoders: 
+    - 'Standard efficient CSS encoders \cite{arXiv:9608026}'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_reed_muller.yml:23]: This should go into encoder statemnt for CSS codes. Remove duplicate in decoders field.

File: codes/quantum/qubits/quantum_reed_muller.yml [hunk 11]
```diff
@@ -5,26 +5,45 @@
...
 
-#notes:
-#  - ''
+  fault_tolerance:
+    - 'Gate switching protocol for universal computation \cite{arxiv:1403.2734}.'
 
-#realizations:
-#  - ''
+  threshold: 
+    - 'Between \(10^{-3}\) and \(10^{-6}\), see \cite{doi:10.1088/2058-9565/abb027}'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_reed_muller.yml:35]: Against what noise? With what decoder? Is decoder noiseless?

File: codes/quantum/qubits/quantum_reed_muller.yml [hunk 12]
```diff
@@ -5,26 +5,45 @@
...
+  
+notes:
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_reed_muller.yml:37]: Move first item to detail field of a cousin entry with code_id: quantum_convolutional
Move second item to detail field of a cousin entry with code_id: triorthogonal
Move third item to rate field

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 13]
```diff
@@ -0,0 +1,46 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: quantum_triorthogonal
+physical: qubits
+logical: qubits
+
+name: '(Quantum) Triorthogonal'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:10]: Quantum triorthogonal code

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+introduced: '\cite{arXiv:1209.2426}'
+
+description: 'A triorthogonal \(m \times n\) binary matrix is triorthogonal if its rows \(r_1, \ldots, r_m\) satisfy \(|r_i \cdot r_j| = 0\) and \(|f_i \cdot f_j \cdot f_k| = 0\), where addition and multiplication are done on \(\mathbb{Z}_2\). The triorthogonal code associated with the matrix is constructed by mapping non-zero entries in even-weight rows to X operators, and Z operators for each row in the orthogonal complement.'
+
+protection: 'Protects against distance 2 Pauli errors \cite{arXiv:1209.2426}'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:16]: You meant weight two? Is the distance always 3? If so, then add first sentence in description that says this is a family of [[n,k,3]] CSS codes constructed using triorthogonal matrices.

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  encoders: 
+    - 'Encoder for magic states for the code constructed in \cite{arXiv:1209.2426}.'
+
+  transversal_gates: 'Admites transversal T gates \cite{arXiv:1209.2426}. Also admits the controlled-controlled-Z gate.'
+
+  threshold: 
+    - 'Approximately \(\frac{1}{3k + 1}\) \cite{arXiv:1209.2426}.'
+  
+realizations:
+  - 'Codes used in magic state distillation.'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:31]: Move this to description. Explain why this is the case if you can.

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+notes:
+  - 'Used to construct protocols for distilling high-quality magic states \cite{arXiv:1209.2426}.'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:34]: Same with this. Move to description.

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+  - '\cite{arXiv:2107.09684} presents a classification of triorthogonal codes up to \(n + k \leq 38\) by associating each triorthogonal code with a Reed-Muller polynomial.'
+
+relations:
+  parents:
+    - code_id: css
+      detail: 'all triorthogonal codes are CSS codes'
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:40]: No detail needed

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+
+  cousins:
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:42]: Is stab_15_1_3 triorthogonal? If so, please add it as cousin.

File: codes/quantum/qubits/quantum_triorthogonal.yml [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+    - code_id: quantum_reed_muller
+      detail: 'classification of triorthongonal yields a connection to Reed-Muller polynomials \cite{arXiv:2107.09684}'
+    - code_id: reed_muller
```
Comment (valbert4) [inline codes/quantum/qubits/quantum_triorthogonal.yml:45]: remove reed_muller

**Comment by valbert4** [comment]
Thanks. Please do a new commit with the files and the suggested changes.

File: codes/quantum/qubits/permutation_invariant.yml (hunk 9 above)
Comment (bquiring) [reply codes/quantum/qubits/permutation_invariant.yml:1]: happy to do so

**Pull request closed without merging**